core/src/test/java/io/confluent/rest/RequestLogHandlerIntegrationTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2023 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import static io.confluent.rest.TestUtils.getFreePort;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.MediaType;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.server.CustomRequestLog;
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+
+@Tag("IntegrationTest")
+public class RequestLogHandlerIntegrationTest {
+
+  private final HttpClient httpClient = new HttpClient();
+  private Server server;
+
+  @Captor
+  private ArgumentCaptor<org.eclipse.jetty.server.Request> requestCaptor;
+
+  @Captor
+  private ArgumentCaptor<org.eclipse.jetty.server.Response> responseCaptor;
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    httpClient.start();
+    requestCaptor = ArgumentCaptor.forClass(org.eclipse.jetty.server.Request.class);
+    responseCaptor = ArgumentCaptor.forClass(org.eclipse.jetty.server.Response.class);
+  }
+
+  @AfterEach
+  public void tearDown() throws Exception {
+    httpClient.stop();
+    if (server != null) {
```

Comment:
Since server is started in the test below, then it should be stopped in test below.

Reply:
for sure